### DISCUSSION TITLE
Convergence Issues for Multiapp Fracture Flow Example with Multiphase Mass Transfer

### DISCUSSION THREAD

**Discussion by zerog981** [idaholab/moose#27212] [Q&A Modules: Porous Flow]
Dear MOOSE experts,
I'm currently trying to adopt the [Multiapp Fracture Flow Example](https://mooseframework.inl.gov/modules/porous_flow/multiapp_fracture_flow_introduction.html) for use in a larger project concerning CO2 injection into supercritical water. I was able to implement mass transfer as suggested in the corresponding section, but I keep running into convergence issues.
All files can be found in the following repository: https://github.com/zerog981/moose-fracture-example

The original example is bundled with Moose ([https://github.com/idaholab/moose/tree/next/modules/porous_flow/examples/multiapp_fracture_flow/3dFracture](https://github.com/idaholab/moose/tree/next/modules/porous_flow/examples/multiapp_fracture_flow/3dFracture)).

Here is a list of variations I tried:
- All different sizes of meshes. 
- Reducing Nonlinear absolute tolerance down to 1e-4 with no success.
- Changing fluid properties to `SimpleFluidProperties`. This does improve the situation, but being in the supercritical region requires more accurate equations of state. I also tried using `TabulatedFluidProperties`, but then I get errors about MOOSE trying to sample out of range.
- If mass transfer is deactivated and only conductive heat transfer is considered, the simulation runs without problems.
- Instead of including the varying fracture porosity from the example, I assumed constant aperture for simplification (knowing that this is quite different in reality).

Before crashing, the exchange of gas into the matrix is clearly visible (after about 170s into the simulation):
![snapshot](https://github.com/idaholab/moose/assets/105823281/26229de8-ba1e-4a59-887c-c8903e6285c4)

I'm still quite new to MOOSE and this is my first project, so any help or guidance is highly appreciated. Please tell me if you need more information.

This is the log:
[failure.log](https://github.com/idaholab/moose/files/14790866/failure.log)

**Comment by GiudGiud** [comment]
Hello

> but then I get errors about MOOSE trying to sample out of range.

did you try to extend the range of the tabulations? also these might not matter if the final converged values are within range. 
The nonlinear solver can visit non-physical regions during the iteration process. 

Can you try some of the techniques in there and report back?
https://mooseframework.inl.gov/moose/application_usage/failed_solves.html

Guillaume

**Reply by zerog981** [reply]
Thank you @GiudGiud for your swift reply, yes, I tried extending the range already to the maximum without success. In the end I was using the EOS objects directly (`CO2FluidProperties` and `Water97FluidProperties`).

> https://mooseframework.inl.gov/moose/application_usage/failed_solves.html
I'm aware of that page, but I went through it again:
- Initial conditions seem to be fine to me
- I tried including `line_search = 'none'` without success
- From [show_var_residual_norms](https://mooseframework.inl.gov/moose/application_usage/failed_solves.html#1f4f80b7-8231-428e-bb96-a2cd914d287f) I saw that temperature residuals are off by 5-6 orders of magnitude (despite `automatic_scaling` being true). So I added temperature scaling of 1e-6 and deactivated automatic scaling. The convergence issue is not affected by this.

Chris

**Reply by GiudGiud** [reply]
did you try reducing the problem to a few thousand DOFs and check if SVD gets you a decent condition number

is the linear solve or the nonlinear solve not converging?

**Reply by zerog981** [reply]
I used a strong machine to not have to reduce the DOFs (cannot edit the fracture network meshing), and I got condition numbers as high as 1e12 (fracture network) and 1e11 (matrix). I suppose that's too high?

Occasionally I see linear solves fail, but this is solved after one timestep cut. The real failure is caused by the nonlinear solve not converging.

As reference for SVD: https://mooseframework.inl.gov/moose/application_usage/failed_solves.html

**Reply by GiudGiud** [reply]
1e12 might be fine actually. Maybe with a customized nonlinear scaling (scaling parameter for each parameter) you might be able to get convergence?
Using `Debug/show_var_residual_norms` can you find a few good scaling parameters for each variable (equation really)
